File: src/my_project/easy_problems/from251to300/reverse_vowels_string.py
from typing import List 

class Solution:
    def reverseVowels(self, s: str) -> str:
        
   
        ind_vec = []
        for index,char in enumerate(s):
            if char in ['a','e','i','o','u','A','E','I','O','U']:
                ind_vec.append(index)
        N = len(ind_vec)
        
        s_copy = [letter for letter in s]
        for i in range(len(ind_vec)):
            s_copy[ind_vec[i]] = s[ind_vec[N-1-i]]
            
        

        
        
        return ''.join(s_copy)

File: my_project_typescript/easy_problems_typescript/from151to200/reverseVowelsString.ts
function reverseVowels(s: string): string {

    let ind_vec: Array<number> = []
    let vowels: Array<string> = ['a','e','i','o','u','A','E','I','O','U']

    for (let i = 0; i < s.length; i++){
        if (vowels.includes(s[i])){
            ind_vec.push(i)
        }
    }

    let N: number = ind_vec.length

    let s_copy: Array<string> = []

    for (let letter of s){
        s_copy.push(letter)
    }

    for (let i = 0; i < ind_vec.length; i++){
        s_copy[ind_vec[i]] = s[ind_vec[N-1-i]]
    }

    return s_copy.join('')
};